homework-selects/sql_questions.sql

```diff
@@ -1,57 +1,841 @@
+SELECT  CONCAT(CONCAT(first_name, ' '), last_name) AS employee_name
+FROM employees
+WHERE salary > (
+        SELECT salary
+        FROM employees
+        WHERE employee_id = 163
+);
+    
 --16. the employee id, employee name (first name and last name ) for all employees who earn more than the average salary.
+SELECT employee_id,
+       CONCAT(CONCAT(first_name, ' '), last_name) AS employee_name
+FROM employees
+WHERE salary > (
+        SELECT AVG(salary)
+        FROM employees
+);
+
 --17. the employee name ( first name and last name ), employee id and salary of all employees who report to Payam.
+SELECT employee_id,
+       CONCAT(CONCAT(first_name, ' '), last_name) AS employee_name,
+       salary
+FROM employees
+WHERE manager_id = (
+        SELECT manager_id
+        FROM employees
+        WHERE first_name = 'Payam'
+);
+
 --18. the department number, name ( first name and last name ), job and department name for all employees in the Finance department.
+SELECT  CONCAT(CONCAT(first_name, ' '), last_name) AS employee_name,
+        department_id,
+        job_title,
+        department_name
+FROM employees
+LEFT JOIN departments USING ( department_id )
+LEFT JOIN jobs USING ( job_id )
+WHERE department_name = 'Finance';
+
 --19. all the information of an employee whose id is any of the number 134, 159 and 183.
+SELECT *
+FROM employees
+WHERE employee_id IN ( 134, 159, 183 );
+
 --20. all the information of the employees whose salary is within the range of smallest salary and 2500.
+SELECT *
+FROM employees
+WHERE salary > (
+        SELECT MIN(salary)
+        FROM employees
+) AND salary < 2500;
+
 --21. all the information of the employees who does not work in those departments where some employees works whose id within the range 100 and 200.
+SELECT
+    *
+FROM
+    employees e1
+WHERE
+    e1.department_id NOT IN (
+        SELECT
+            e2.department_id
+        FROM
+            employees e2
+        WHERE
+            e1.employee_id BETWEEN 100 AND 200
+    );
+
 --22. all the information for those employees whose id is any id who earn the second highest salary.
+SELECT
+    *
+FROM
+    employees
+WHERE
+    salary = (
+        SELECT
+            salary
+        FROM
+            (
+                SELECT
+                    s.salary,
+                    ROWNUM rm
+                FROM
+                    (
+                        SELECT DISTINCT
+                            salary
+                        FROM
+                            employees
+                        ORDER BY
+                            salary DESC
+                    ) s
+                WHERE
+                    ROWNUM <= 2
+            )
+        WHERE
+            rm >= 2
+    );
+
+
 --23. the employee name( first name and last name ) and hiredate for all employees in the same department as Clara. Exclude Clara.
+SELECT
+    e1.first_name,
+    e1.last_name,
+    start_date
```

Comment:
you must retrieve hire_date, not start_date. hire_date is one of the columns from employees table